lib/ruby_lsp/requests/on_type_formatting.rb

```diff
@@ -69,10 +69,7 @@ def handle_pipe
 
         # If the current character is a pipe and both previous ones are pipes too, then we autocompleted a pipe and the
         # user inserted a third one. In this case, we need to avoid adding a fourth and remove the previous one
-        if line[@position[:character] - 2] == "|" &&
-            line[@position[:character] - 1] == "|" &&
-            line[@position[:character]] == "|"
-
+        if line[..@position[:character]] =~ /do\s+\|[^|]*?\|\|$/
```

Comment:
Will this work for one line blocks? E.g.:
```ruby
[].each { || }
```

Also, the comments above this if statement have to be updated given that they no longer reflect the implementation.